app/scripts/views/complete_reset_password.js

```diff
@@ -17,17 +18,42 @@ define(function (require, exports, module) {
   const Template = require('templates/complete_reset_password.mustache');
   const Url = require('../lib/url');
   const VerificationInfo = require('../models/verification/reset-password');
+  const AccountRecoveryVerificationInfo = require('../models/verification/account-recovery');
 
   const proto = FormView.prototype;
   const View = FormView.extend({
     template: Template,
     className: 'complete-reset-password',
 
+    events: _.extend({}, FormView.prototype.events, {
+      'click .remember-password': FormView.preventDefaultThen('_navigateToSignin')
+    }),
+
+    _navigateToSignin: function () {
+      this.navigate('signin');
+    },
+
     initialize (options) {
       options = options || {};
 
       var searchParams = Url.searchParams(this.window.location.search);
       this._verificationInfo = new VerificationInfo(searchParams);
+
+      const model = options.model;
+
+      // If this property is set, this will ensure that a regular password reset
+      // is preformed by *not* setting any `recoveryKeyId` data.
+      this.lostRecoveryKey = model && model.get('lostRecoveryKey');
+      if (this.lostRecoveryKey) {
+        return;
+      }
+
+      // If the complete password screen was navigated from the account recovery confirm
+      // key view, then these properties must be set in order to recover the account
+      // using the recovery key.
+      if (model && options.model.get('recoveryKeyId')) {
+        this._accountRecoveryVerficationInfo = new AccountRecoveryVerificationInfo(options.model.toJSON());
```

Comment:
No biggie, but these two lines still dereference the full `options.model` even though you locally aliased it to `model` above.